### PR TITLE
RiiR cert updater

### PR DESCRIPTION
**Pull request by tshepang**
Part of #39

### CODE REVIEW

File: src/tests/verification_real_world/update_valid_ee_certs.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env cargo +nightly -Z script
```
Comment (cpu) [inline src/tests/verification_real_world/update_valid_ee_certs.rs:1]: Nice! I've been meaning to experiment with cargo scripts but haven't had a chance. This seems like an excellent use-case.

I'm having some trouble running this, and I'm not sure if it's my fault or not. Do you have any ideas?

```
$> ./update_valid_ee_certs.rs
error: toolchain 'nightly -Z script' is not installed
```

```
$ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/daniel/.rustup

installed toolchains
--------------------

nightly-x86_64-unknown-linux-gnu (default)
1.60-x86_64-unknown-linux-gnu
1.72-x86_64-unknown-linux-gnu

installed targets for active toolchain
--------------------------------------

aarch64-linux-android
i686-unknown-linux-gnu
wasm32-unknown-unknown
x86_64-linux-android
x86_64-unknown-linux-gnu

active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.77.0-nightly (3d0e6bed6 2023-12-21)
```

File: src/tests/verification_real_world/update_valid_ee_certs.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+```cargo
+package.edition = "2021"
+dependencies.anyhow = "1"
+dependencies.reqwest.version = "0.11"
+dependencies.reqwest.features = ["blocking"]
```
Comment (cpu) [inline src/tests/verification_real_world/update_valid_ee_certs.rs:6]: Will this example work if we also specify the rustls-tls and webpki-roots features, or is it dependent on native-tls?

**Review by cpu — COMMENTED** [review]
Thanks for picking this up!

File: src/tests/verification_real_world/update_valid_ee_certs.rs (hunk 1 above)
Comment (tshepang) [reply src/tests/verification_real_world/update_valid_ee_certs.rs]: I suspect b74940aba1bf6dd723ba651b87302931cf0b30ce fixes it. My macOS did not need this change, and I guess Linux does.

File: src/tests/verification_real_world/update_valid_ee_certs.rs (hunk 2 above)
Comment (tshepang) [reply src/tests/verification_real_world/update_valid_ee_certs.rs]: 5ce3fefa2c7e796a52d6a130812d1c2682d8b1a9 should do it

**Comment by tshepang** [comment]
as a sidenote, lencr.org has a new cert

File: src/tests/verification_real_world/update_valid_ee_certs.rs (hunk 1 above)
Comment (cpu) [reply src/tests/verification_real_world/update_valid_ee_certs.rs]: That did the trick, thanks :+1:

File: src/tests/verification_real_world/update_valid_ee_certs.rs (hunk 2 above)
Comment (cpu) [reply src/tests/verification_real_world/update_valid_ee_certs.rs]: Awesome :+1:

**Review by cpu — APPROVED** [review]
Looks great, thank you. Would you mind squashing your commits into one?

**Comment by tshepang** [comment]
I done it

File: src/tests/verification_real_world/update_valid_ee_certs.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    let response = reqwest::blocking::Client::builder()
+        .tls_info(true)
+        // avoids agilebits.com redirect, which will result in the wrong cert...
+        // we want the cert of agilebits.com, not of 1password.com
+        .redirect(reqwest::redirect::Policy::none())
+        .build()?
+        .get(url)
+        .send()?;
+    let tls_info: Option<&reqwest::tls::TlsInfo> = response.extensions().get();
+    if let Some(tls_info) = tls_info {
```
[the remark this answers is not in the document]
Comment (djc) [reply src/tests/verification_real_world/update_valid_ee_certs.rs:34]: I'd propose writing it like this:

```rust
let Some(tls_info) = response.extensions().get::<&reqwest::tls::TlsInfo>() else {
    anyhow::bail!("no TLS info found");
};

.. // similar for getting the peer_certificate()`
```

[the remark this answers is not in the document]
Comment (tshepang) [reply src/tests/verification_real_world/update_valid_ee_certs.rs]: https://github.com/rustls/rustls-platform-verifier/pull/44/commits/1a9d5cec9e3ad0979f25e493b268b433dd5112b8

**Review by cpu — APPROVED** [review]

**Pull request merged**